fv3core/stencils/xtp_u.py

```diff
@@ -123,37 +122,46 @@ def _compute_stencil(
         flux = _get_flux(u, courant, rdx, bl, br)
 
 
-def compute(c: FloatField, u: FloatField, flux: FloatField):
-    """
-    Compute flux of kinetic energy in x-dir.
-
-    Args:
-        c (in): Courant number in flux form
-        u (in): x-dir wind on D-grid
-        flux (out): Flux of kinetic energy
-    """
-    grid = spec.grid
-    iord = spec.namelist.hord_mt
-    if iord not in (5, 6, 7, 8):
-        raise NotImplementedError(
-            "Currently xtp_v is only supported for hord_mt == 5,6,7,8"
+class XTP_U:
+    def __init__(self, namelist):
+        iord = spec.namelist.hord_mt
+        if iord not in (5, 6, 7, 8):
+            raise NotImplementedError(
+                "Currently xtp_v is only supported for hord_mt == 5,6,7,8"
+            )
+        self.grid = spec.grid
+        self.origin = self.grid.compute_origin()
+        self.domain = self.grid.domain_shape_compute(add=(1, 1, 0))
+        ax_offsets = axis_offsets(self.grid, self.origin, self.domain)
+        assert namelist.grid_type < 3
+        self.stencil = gtscript.stencil(
+            definition=_compute_stencil,
+            externals={
+                "iord": iord,
+                "mord": iord,
+                "xt_minmax": False,
+                **ax_offsets,
+            },
+            backend=global_config.get_backend(),
+            rebuild=global_config.get_rebuild(),
         )
 
-    stencil = gtstencil(
-        definition=_compute_stencil,
-        externals={
-            "iord": iord,
-            "mord": iord,
-            "xt_minmax": False,
-        },
-    )
-    stencil(
-        c,
-        u,
-        flux,
-        grid.dx,
-        grid.dxa,
-        grid.rdx,
-        origin=grid.compute_origin(),
-        domain=grid.domain_shape_compute(add=(1, 1, 0)),
-    )
+    def __call__(self, c: FloatField, u: FloatField, flux: FloatField):
+        """
+        Compute flux of kinetic energy in x-dir.
+
+        Args:
+            c (in): Courant number in flux form
+            u (in): x-dir wind on D-grid
+           flux (out): Flux of kinetic energy
```

Comment:
So clean!

Misaligned tab:
```suggestion
        Args:
            c (in): Courant number in flux form
            u (in): x-dir wind on D-grid
            flux (out): Flux of kinetic energy
```